router/proxy/reverseproxy.go

```diff
@@ -128,22 +128,32 @@ func (p *ReverseProxy) serveUpgrade(rw http.ResponseWriter, req *http.Request) {
 	}
 	defer uconn.Close()
 
-	p.writeResponse(rw, res)
-	res.Body.Close()
+	dconn, bufrw, err := rw.(http.Hijacker).Hijack()
```

Comment:
yeah, I think it would, because it would need 2 blocks for sending back the response.